tests/reftests/list-repo.test

```diff
@@ -0,0 +1,356 @@
+7371c1d9
```

Comment:
this test was added in https://github.com/ocaml/opam/pull/5006 to have some tests for opam list https://github.com/ocaml/opam/pull/5005. I chose lacaml because it had a reasonable number of revdeps and recursive deps, as well as depopts. Feel free to change the test so it shows that but i don't think it's worth the time investment at the moment

Reply:
I see. Now that we have benches, we could move the big depend on and recursive deps test on bench, and have a small feature focused on for reftest. But it's not the purpose of this PR.
I've added #6140 to remember it.
As it is moved, it worth renaming it more purposely, like list-big-dependencies, or something like that.

Reply:
done